File: bsp4j/src/ch/epfl/scala/bsp4j/RustBuildServer.java
package ch.epfl.scala.bsp4j;

import java.util.concurrent.CompletableFuture;
import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;

public interface RustBuildServer {
  @JsonRequest("buildTarget/rustWorkspace")
  CompletableFuture<RustWorkspaceResult> rustWorkspace(RustWorkspaceParams params);

  @JsonRequest("buildTarget/rustToolchain")
  CompletableFuture<RustToolchainResult> rustToolchain(RustToolchainParams params);
}

File: bsp4kt/src/org/jetbrains/bsp/bsp4kt/RustBuildServer.kt
package org.jetbrains.bsp.bsp4kt

import org.jetbrains.jsonrpc4kt.services.JsonNotification
import org.jetbrains.jsonrpc4kt.services.JsonRequest
import kotlinx.serialization.SerialName

interface RustBuildServer {
  @JsonRequest("buildTarget/rustWorkspace")
  suspend fun rustWorkspace(params: RustWorkspaceParams): RustWorkspaceResult

  @JsonRequest("buildTarget/rustToolchain")
  suspend fun rustToolchain(params: RustToolchainParams): RustToolchainResult

}
